.github/workflows/publish-to-pypi.yml

```diff
@@ -5,22 +5,37 @@ on:
     types: [released]
 
 jobs:
-  build:
-      name: Publish release to PyPI
-      env:
-        PYPI_USERNAME_STSCI_MAINTAINER: ${{ secrets.PYPI_USERNAME_STSCI_MAINTAINER }}
-        PYPI_PASSWORD_STSCI_MAINTAINER: ${{ secrets.PYPI_PASSWORD_STSCI_MAINTAINER }}
-        PYPI_USERNAME_OVERRIDE: ${{ secrets.PYPI_USERNAME_OVERRIDE }}
-        PYPI_PASSWORD_OVERRIDE: ${{ secrets.PYPI_PASSWORD_OVERRIDE }}
-        PYPI_TEST: ${{ secrets.PYPI_TEST }}
-        INDEX_URL_OVERRIDE: ${{ secrets.INDEX_URL_OVERRIDE }}
-      runs-on: ubuntu-latest
-      steps:
+  validate:
+    name: Validate metadata
+    runs-on: ubuntu-latest
+    steps:
+      - uses: spacetelescope/action-publish_to_pypi/validate@master
 
-          # Check out the commit containing this workflow file.
-          - name: checkout repo
-            uses: actions/checkout@v2
-         
-          - name: custom action
-            uses: spacetelescope/action-publish_to_pypi@master
-            id: custom_action_0
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    needs: [validate]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, macos-10.15]
+
```

Comment:
Currently we build wheels for [all of the versions of Python that `cibuildwheel` builds for by default](https://github.com/pypa/cibuildwheel#what-does-it-do), so CPython 3.6-3.10, plus PyPy 3.7 v7.3.

I imagine we probably want the versions of Python we support to be an organization-wide decision, and thus defined inside of our custom step. If teams want to include/exclude specific versions for their packages though, they can add an `env` block to the `spacetelescope/build-wheel` step and use [the `CIBW_BUILD` and `CIBW_SKIP` variables](https://cibuildwheel.readthedocs.io/en/stable/options/#build-skip) to modify the behavior.

Reply:
Also, is there a reason to not use `macos-latest` and `ubuntu-latest`?  Since `stsci.imagestats` is a bit of an ancient package that doesn't get released often (ever?) except to fix build issues, those general pointers might be better to use than a specific version?

Might we also want to build on macos with Apple arm64 too?  Is this better to specified in a per-repo configuration file, i.e. like

https://cibuildwheel.readthedocs.io/en/stable/options/#configuration-file

so as to keep the actual workflows in all the repos standardized?  Or better set in the workflow?

Reply:
No particular reason, these versions were copied from the `cibuildwheel` examples, so updating them to `-latest` might be a good idea.

I suppose we could indeed populate the build matrix from a configuration file actually, that's a pretty great idea. I imagine we'd want to default to an organization-wide setting if the file isn't present. Avoiding the need for developers to touch the workflow file would be awesome.

Actually, now that I think of it, it might be possible for us to define the build matrix as the output of another custom step, thus allowing us to move the definition into [`spacetelescope/action-publish_to_pypi`](https://github.com/spacetelescope/action-publish_to_pypi/), making it easier for us to add/modify supported platforms in the future.

Reply:
Cool.  Stuff to think about.  I think currently this is great and a big improvement over what we are doing now - not building wheels.  =)